test/functional/core/startup_spec.lua

```diff
@@ -458,7 +458,7 @@ describe('user config init', function()
 
   it('loads init.lua from XDG config home by default', function()
     clear{ args_rm={'-u' }, env={ XDG_CONFIG_HOME=xconfig }}
-
+    feed('<cr>') -- confirm "Conflicting config ..." message
```

Comment:
hmm, in this test, such message should not be exist on clear setup(`init.vim` is not created in this context).

Reply:
how did you run the tests? does test `Xhome` directory get removed between test runs?

Reply:
I just ran with make test. Let me see if adding this back causes the test to hang. When I tested before, I had to disable this test in order for the test suite to not hang.

Reply:
If I remove that one, it hangs.

Reply:
I was unable to reproduce it, try to debug on your side.
for example, put some "test" expectation after `feed`
```
feed('<cr>') 
matches('check it!', meths.exec('messages', true))
```

Reply:
Not sure if helpful, but this test was hanging for me as well. I tried the debugging suggestion and got the following output:

```
[ RUN      ] user config init loads init.lua from XDG config home by default: 1020.91 ms ERR
test/helpers.lua:100: Pattern does not match.
Pattern:
check it!
Actual:
E484: Can't open file $HOME/bin/share/nvim/syntax/syntax.vim

stack traceback:
        test/helpers.lua:100: in function 'matches'
        test/functional/core/startup_spec.lua:463: in function <test/functional/core/startup_spec.lua:459>
```

I was surprised to see it was referencing `$HOME/bin`. I had run `make CMAKE_INSTALL_PREFIX=$HOME/bin/` to build but hadn't run `make install` yet. Running `make install` fixed the hanging test. Unsure if this is working as intended.

Reply:
yeah, there is certainly a problem with setup of testing environment(maybe `XDG_CONFIG_HOME` somehow leaked)
I'm going to take a look at this one more time.

Reply:
@dm1try any news ? I am hit by this too. Error is triggered when neovim sees both an init.lua and init.vim but I haven't looked into why it finds both (leftover from previous state ? )

Reply:
>  any news ?

@teto I checked the assumption above(about `XDG_CONFIG_HOME`) but didn't find anything suspicious.

> Error is triggered when neovim sees both an init.lua and init.vim but I haven't looked into why it finds both (leftover from previous state ? )

if this is still reproducible , could you re-check if `XDG_CONFIG_HOME` points to "Xhome/Xconfig"
and is this directory removed on each test run(maybe there is some os-specific behavior and [this setup](https://github.com/neovim/neovim/blob/2dd383c943ea31abf30ec1e39fbeb5f5971131aa/test/functional/core/startup_spec.lua#L442-L457) does not work as expected)
 
also, maybe the tests are run in parallel somehow?)

Reply:
Having some trouble reproducing the hangups